manifests/repos.pp

```diff
@@ -25,15 +25,20 @@
       }
       Exec['apt_update'] -> Package[$docker::prerequired_packages]
       if ($docker::use_upstream_package_source) {
+        ensure_packages(['debian-keyring', 'debian-archive-keyring'])
         apt::source { 'docker':
-          location          => $location,
-          release           => $docker::package_release,
-          repos             => $docker::package_repos,
-          key               => $package_key,
-          key_source        => $key_source,
-          required_packages => 'debian-keyring debian-archive-keyring',
-          pin               => '10',
-          include_src       => false,
+          location => $location,
+          release  => $docker::package_release,
+          repos    => $docker::package_repos,
+          key      => {
+            'id'     => $package_key,
+            'server' => 'hkp://keyserver.ubuntu.com:80',
+          },
+          pin      => '10',
+          require  => [
```

Comment:
the `requires` here exacerbates the issue discussed in #369  

also note https://github.com/puppetlabs/puppetlabs-apt/pull/427#issuecomment-75166159



> The idea here was that for some repositories you need a package installed that provides the keys for the repo. In the  case of the main Debian repositories, the debian(-archive)-keyring packages.
>
However, if you manage to install a Debian system without those packages you deserve what's coming to you, a world of pain. 

I suggest not managing those packages in this puppet-module.